File: app/src/main/java/com/lelloman/demo/fragment/GithubIdenticonFragment.java
package com.lelloman.demo.fragment;

import com.lelloman.identicon.IdenticonDrawable;
import com.lelloman.identicon.github.GithubIdenticonDrawable;


public class GithubIdenticonFragment extends IdenticonFragment {

	@Override
	protected RecyclerViewAdapter getAdapter() {
		return new IdenticonFragment.RecyclerViewAdapter() {
			@Override
			protected int getHashForPosition(int position) {
				return RANDOM.nextInt();
			}

			@Override
			protected IdenticonDrawable makeIdenticonDrawable(int width, int height, int hash) {
				return new GithubIdenticonDrawable(width, height, hash);
			}
		};
	}
}

File: app/src/main/java/com/lelloman/demo/fragment/GithubIdenticonFragment.kt
package com.lelloman.demo.fragment

import com.lelloman.identicon.drawable.GithubIdenticonDrawable
import com.lelloman.identicon.drawable.IdenticonDrawable

class GithubIdenticonFragment : IdenticonFragment() {

    override val adapter: IdenticonFragment.RecyclerViewAdapter
        get() = object : IdenticonFragment.RecyclerViewAdapter() {
            override fun getHashForPosition(position: Int): Int {
                return IdenticonFragment.RANDOM.nextInt()
            }

            override fun makeIdenticonDrawable(width: Int, height: Int, hash: ByteArray): IdenticonDrawable {
                return GithubIdenticonDrawable(width, height, hash)
            }
        }
}
